src/octoprint/util/comm.py

```diff
@@ -3126,134 +3184,213 @@ def _enqueue_for_sending(self, command, linenumber=None, command_type=None, on_s
 			self._logger.debug("Type already in send queue: " + e.type)
 			return False
 
-	def _send_loop(self):
+	def _pool_for_freebuffer(self):
```

Comment:
Please rename to `_wait_for_free_buffer` or something. `pool_for_freebuffer` really just makes me scratch my head and doesn't tell me anything.